### PR TITLE
New blog post. Deploying Torrust To Production

### LINKED ISSUE #103: New article: Deploying Torrust To Production
My idea is to write a step-by-step tutorial that people can follow to deploy the **Torrust Full Solution** (Tracker + Index + Index GUI) to a virtual machine using docker and docker compose.

There are many options for deployments but I want to do the most generic and basic one so that it can be reused in any hosting provided.

### What is included

- Deploy to a Digital Ocean virtual machine (droplet).
- Use docker compose.
- Expose the Tracker and Index GUI.
- Use HTTPS for all the services with Let's Encrypt.
- Automatic certificate renewal.
- Use SQLite.
- Tracker will be configured as public.

### What is NOT included

- Database backups configuration.
- DNS setup.
- Security. I'm not an expert in security, so you must be sure you set up your networks, firewalls, services correctly.
- SMTP configuration. Email validation on sigup will be not required.

### Why this concrete example

- It's generic, if you know how to do this you can deploy to any provider. I only want to explain the parts you need to put together. I'm not worried about performance, automation, etcetera.

### Links

The following tutorial is more or less what I want to do:

https://www.digitalocean.com/community/tutorials/how-to-secure-a-containerized-node-js-application-with-nginx-let-s-encrypt-and-docker-compose

but specific with Torrust, so you can copy/paste all the commands.

### Previous work

I did some experiments in the past to run Torrust with different providers and configurations:

https://github.com/josecelano/awesome-torrust-tracker-compose

### Extra

We should find a good name for **Torrust Full Solution** (Tracker + Index + Index GUI). I suppose it could be called "Torrust Index" but there is a conflict with the actual [Torrust Index](https://github.com/torrust/torrust-index).

I think we need a good name to avoid confusion and because in the future we could have other solutions.

### Future work

I would like to automatic this process for at least one provider so users can "click & use".

### Live demo

I will use these domains:

- https://index.torrust-demo.com
- https://tracker.torrust-demo.com

I will keep the demo live during the Beta phase so users can test it.

cc @da2ce7 @WarmBeer @mario-nt @grmbyrn @cgbosse

### PR DESCRIPTION
**Pull request by josecelano**
A new blog post explaining how to deploy Torrsut to a virtual machine with Docker Compose.

### CODE REVIEW

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md [hunk 1]
```diff
@@ -0,0 +1,1073 @@
...
+After cloning the repository you have to run the install script:
+
+<CodeBlock lang="bash">
+
+```bash
+$ ./bin/install.sh 
+Creating compose .env './.env'
+Creating proxy config file: './storage/proxy/etc/nginx-conf/nginx.conf'
+Creating index database: './storage/index/lib/database/sqlite3.db'
+Crating index configuration: './storage/index/etc/index.toml'
```
Comment (cgbosse) [inline src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: Creating?

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md [hunk 2]
```diff
@@ -0,0 +1,1073 @@
...
+Creating tracker database: './storage/tracker/lib/database/sqlite3.db'
+Crating tracker configuration: './storage/tracker/etc/tracker.toml'
```
Comment (cgbosse) [inline src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: Creating?

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md [hunk 3]
```diff
@@ -0,0 +1,1073 @@
...
+- Tracker (HTTP): <https://tracker.torrust-demo.com/announce>
+- Tracker (API): <https://tracker.torrust-demo.com/api>
+
+You can disable both, direct access to the container (without using the Nginx proxy) and also access without certificates (HTTP). In that case you only need to expose port: `443` and `6969`. In order to do that you will
+need to enable a firewall on the server or the Digital Ocean firewall.
+
+<Callout type="info">
+  Notice: The UDP tracker is not available via the proxy. If you want all requests to go through the
+  proxy you need to change the Nginx configuration to set up a UDP proxy. There is a discussion
+  on the Tracker repository on GitHub about about how to setup the proxy for the UDP tracker.
```
Comment (cgbosse) [inline src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: Duplicated word, "about"

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md [hunk 4]
```diff
@@ -0,0 +1,1073 @@
...
+
+</CodeBlock>
+
+<Callout type="info">
+  However, there is a problem with setting up your own firewall. By default, Docker manipulates iptables to set up network isolation. However, this can interfere with UFW. To fix this, you need to tell Docker not to modify iptables.
+</Callout>
+
+### Health checks
+
+Some services expose public health checks endpoints.
```
Comment (cgbosse) [inline src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: health check endpoints

Check should be singular

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md [hunk 5]
```diff
@@ -0,0 +1,1073 @@
...
+
+    ...
+}
+```
+
+</CodeBlock>
+
+If your server does not have too much disk space you could run out of disk space.
+
+Docker stores the logs in a separate directory, if you wan to truncate the logs you can find them first with:
```
Comment (cgbosse) [inline src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: if you want to truncate

you forgot a "t"

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md [hunk 6]
```diff
@@ -0,0 +1,1073 @@
...
+
+We hope you find this guide useful. Let us know what you think about it!
+
+If you see something wrong or you want to contribute by:
+
+- Adding new sections.
+- Fixing typos.
+- Making it clearer.
+- Adding links.
+- Or whatever you think it could be interesting to have,
```
Comment (cgbosse) [inline src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: "It" seems not the right word here, I think "that" is more appropriate

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md [hunk 7]
```diff
@@ -0,0 +1,1073 @@
...
+
+please open an [issue](https://github.com/torrust/torrust-website/issues) or a [PR](https://github.com/torrust/torrust-website/pulls).
+
+If you have any questions or issues please open an issue on the corresponding repository:
```
Comment (cgbosse) [inline src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: I think here "in" is the correct preposition rather then "on"

To open an issue in a repository.....

**Review by cgbosse — CHANGES_REQUESTED** [review]
A frew typos and improvements.

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md (hunk 1 above)
Comment (josecelano) [reply src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: Fixed here: https://github.com/torrust/torrust-compose/commit/1d7c3f40794dfea750832fef189e3147ad10c961

Comment (josecelano) [reply src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: Fixed

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md (hunk 5 above)
Comment (josecelano) [reply src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: Fixed

File: src/routes/(blog-article)/deploying-torrust-to-production/+page.md (hunk 7 above)
Comment (josecelano) [reply src/routes/(blog-article)/deploying-torrust-to-production/+page.md]: Fixed

**Comment by josecelano** [comment]
Hi @WarmBeer @da2ce7 this is ready to review. After releasing version `v3.0.0` I will update this article to use the corresponding docker image.

**Comment by josecelano** [comment]
ACK 034c5d93ebda1fd40c4b34bff2e4f1941653dc55

**Pull request merged**